### PR TITLE
✨ Source MongoDB Internal POC: Initial commit of new MongoDB source

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
## What
* Skeleton project for internal POC MongoDB source

## How
* Implement initial project structure for the new connector
* Purposely did **NOT** implement the following:
    * Unit tests
    * Acceptance tests
    * Entries in OSS/Cloud registry
    * Release notes
    * Source definition/specs

## Recommended reading order
1. `MongoDbSource.java`
2. `Dockerfile`
3. `build.gradle`
4. `metadata.yml`
5. `spec.json`

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-v3 test report (commit [`f989c30d88`](https://github.com/airbytehq/airbyte/pull/28875/commits/f989c30d8818dc88db2d188cb537d62468cf4cda)) - ❌

⏲️  Total pipeline duration: 02mn47s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-v3/metadata.yaml | ✅        |
| Connector version semver check                                           | ✅        |
| Connector version increment check                                        | ✅        |
| QA checks                                                                | ❌        |
| Build connector tar                                                      | ✅        |
| Build source-mongodb-v3 docker image for platform linux/x86_64           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_source-mongodb-v3/1690830970/f989c30d8818dc88db2d188cb537d62468cf4cda/source-mongodb-v3/0.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-v3 test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-v3 test report (commit [`b862259b8f`](https://github.com/airbytehq/airbyte/pull/28875/commits/b862259b8f838e3eae44cacca5d7974b093493da)) - ✅

⏲️  Total pipeline duration: 02mn20s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_source-mongodb-v3/1690832173/b862259b8f838e3eae44cacca5d7974b093493da/source-mongodb-v3/0.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-v3 test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-v3 test report (commit [`7f27837af8`](https://github.com/airbytehq/airbyte/pull/28875/commits/7f27837af8c1636d4c553160f8a165ac3fb6ca84)) - ❌

⏲️  Total pipeline duration: 03mn35s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-v3/metadata.yaml | ✅        |
| Connector version semver check                                           | ✅        |
| Connector version increment check                                        | ✅        |
| QA checks                                                                | ❌        |
| Build connector tar                                                      | ✅        |
| Build source-mongodb-v3 docker image for platform linux/x86_64           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_source-mongodb-v3/1690833845/7f27837af8c1636d4c553160f8a165ac3fb6ca84/source-mongodb-v3/0.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-v3 test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-v3 test report (commit [`03395ebd45`](https://github.com/airbytehq/airbyte/pull/28875/commits/03395ebd45d8704a8da478919c21890849110f63)) - ✅

⏲️  Total pipeline duration: 02mn16s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_source-mongodb-v3/1690834677/03395ebd45d8704a8da478919c21890849110f63/source-mongodb-v3/0.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-v3 test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-v3 test report (commit [`36d3c0e280`](https://github.com/airbytehq/airbyte/pull/28875/commits/36d3c0e280bf878f5b1816b05c96025d8120014f)) - ❌

⏲️  Total pipeline duration: 05mn22s 

| Step                                                                         | Result   |
|:-----------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-v3/metadata.yaml     | ✅        |
| Connector version semver check                                               | ✅        |
| Connector version increment check                                            | ✅        |
| QA checks                                                                    | ❌        |
| Build connector tar                                                          | ✅        |
| Build source-mongodb-v3 docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mongodb-v3:integrationTest | ✅        |
| Acceptance tests                                                             | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_source-mongodb-v3/1690834928/36d3c0e280bf878f5b1816b05c96025d8120014f/source-mongodb-v3/0.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-v3 test
```

File: airbyte-integrations/connectors/source-mongodb-v3/acceptance-test-config.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# See [Connector Acceptance Tests](https://docs.airbyte.com/connector-development/testing-connectors/connector-acceptance-tests-reference)
+# for more information about how to configure these tests
+connector_image: airbyte/source-mongodb-v2:dev
```
Comment (colesnodgrass) [inline airbyte-integrations/connectors/source-mongodb-v3/acceptance-test-config.yml:3]: shouldn't this be `v3`?

Comment (jdpgrailsdev) [reply airbyte-integrations/connectors/source-mongodb-v3/acceptance-test-config.yml]: @colesnodgrass Yes...good catch. Now its `-internal-poc`.

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`782afaa13f`](https://github.com/airbytehq/airbyte/pull/28875/commits/782afaa13f0fe32845a0f5411cd36065fd88a636)) - ✅

⏲️  Total pipeline duration: 6.07s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_source-mongodb-v3/1690892989/782afaa13f0fe32845a0f5411cd36065fd88a636/source-mongodb-internal-poc/0.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`e9e887dd7c`](https://github.com/airbytehq/airbyte/pull/28875/commits/e9e887dd7c8a57c8262446c0e77c19ab8e9e1bec)) - ❌

⏲️  Total pipeline duration: 02mn14s 

| Step                                                                               | Result   |
|:-----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-internal-poc/metadata.yaml | ✅        |
| Connector version semver check                                                     | ✅        |
| Connector version increment check                                                  | ✅        |
| QA checks                                                                          | ❌        |
| Build connector tar                                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_source-mongodb-v3/1690893092/e9e887dd7c8a57c8262446c0e77c19ab8e9e1bec/source-mongodb-internal-poc/0.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`d1dc856989`](https://github.com/airbytehq/airbyte/pull/28875/commits/d1dc85698922d6736fef4b4975988f1928d94824)) - ❌

⏲️  Total pipeline duration: 01mn28s 

| Step                                                                               | Result   |
|:-----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-internal-poc/metadata.yaml | ✅        |
| Connector version semver check                                                     | ✅        |
| Connector version increment check                                                  | ✅        |
| QA checks                                                                          | ❌        |
| Build connector tar                                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_source-mongodb-v3/1690894949/d1dc85698922d6736fef4b4975988f1928d94824/source-mongodb-internal-poc/0.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Review by jpefaur — APPROVED** [review]
left one comment

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`61416c82e5`](https://github.com/airbytehq/airbyte/pull/28875/commits/61416c82e5100c76e47c0592aecac80927a7f188)) - ❌

⏲️  Total pipeline duration: 06mn51s 

| Step                                                                                   | Result   |
|:---------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-internal-poc/metadata.yaml     | ✅        |
| Connector version semver check                                                         | ✅        |
| Connector version increment check                                                      | ✅        |
| QA checks                                                                              | ❌        |
| Build connector tar                                                                    | ✅        |
| Build source-mongodb-internal-poc docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mongodb-internal-poc:integrationTest | ✅        |
| Acceptance tests                                                                       | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/jonathan_source-mongodb-v3/1690905888/61416c82e5100c76e47c0592aecac80927a7f188/source-mongodb-internal-poc/0.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Review by colesnodgrass — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`8db2ed9bf9`](https://github.com/airbytehq/airbyte/pull/28875/commits/8db2ed9bf9e0f4ba65590789f577e2ce4f864860)) - ❌

⏲️  Total pipeline duration: 06mn57s 

| Step                                                                                   | Result   |
|:---------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-internal-poc/metadata.yaml     | ✅        |
| Connector version semver check                                                         | ✅        |
| Connector version increment check                                                      | ✅        |
| QA checks                                                                              | ❌        |
| Build connector tar                                                                    | ✅        |
| Build source-mongodb-internal-poc docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mongodb-internal-poc:integrationTest | ✅        |
| Acceptance tests                                                                       | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/jonathan_source-mongodb-v3/1690908839/8db2ed9bf9e0f4ba65590789f577e2ce4f864860/source-mongodb-internal-poc/0.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Comment by octavia-approvington** [comment]
Our work here is done <br/> done

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**